shotover/src/transforms/query_counter.rs

```diff
@@ -26,11 +29,25 @@ pub struct QueryCounterConfig {
 
 impl QueryCounter {
     pub fn new(counter_name: String) -> Self {
-        counter!("shotover_query_count", "name" => counter_name.clone());
+        let counter_name_ref: &'static str = counter_name.leak();
+        let mut counters = HashMap::new();
+        let query_counter = counter!("shotover_query_count", "name" => counter_name_ref);
+        counters.insert("shotover_query_count".to_string(), query_counter);
```

Comment:
We dont actually need to insert this into counters since we never increment this exact counter.
We do want to continue creating it though because it ensures that shotover_query_count shows up with 0 on shotover startup, before any messages have passed through.